### PR TITLE
Petsmain screen challenge UI/yefim yakunin

### PR DESCRIPTION
**Pull request by fimillion**

### CODE REVIEW

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import UIKit
+
+class YeYa_PetCell: UICollectionViewCell {
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:3]: make it final

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:4]: remove space

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+class YeYa_PetCell: UICollectionViewCell {
+    
+    @IBOutlet private weak var shadowView: UIView!
+    @IBOutlet private weak var mainView: UIView!
+    @IBOutlet private weak var petImage: UIImageView!
+    @IBOutlet private weak var nameLabel: UILabel!
+    @IBOutlet private weak var breedLabel: UILabel!
+    @IBOutlet private weak var ageLabel: UILabel!
+    
+    private func setupLayout() {
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:12]: move private methods after all other
Order of methods:
1. override methods
2. internal methods
3. private methods

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+final class YeYa_PetFilterCollectionViewCell: UICollectionViewCell {
+
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift:4]: remove space

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        super.init(frame: frame)
+        setupLayout()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setupLayout()
+    }
+    
+    private func setupLayout() {
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift:18]: order of methods

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/YeYa_PetsViewController.swift [hunk 6]
```diff
@@ -1,8 +1,173 @@
 import UIKit
 
-class YeYa_PetsViewController: UIViewController {
-
+final class YeYa_PetsViewController: UIViewController {
+    @IBOutlet private weak var filtersCollectionView: UICollectionView!
+    @IBOutlet private weak var petsCollectionView: UICollectionView!
+    var pets: [YeYa_PetModel] = []
```
Comment (alexnikol) [inline UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/YeYa_PetsViewController.swift:6]: make all properties that you can as private

**Review by alexnikol — APPROVED** [review]
Fix small changes, and can merge it!

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift (hunk 1 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:3]: final class YeYa_PetCell: UICollectionViewCell {

Added final

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift (hunk 2 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:4]: final class YeYa_PetCell: UICollectionViewCell {
    @IBOutlet private weak var shadowView: UIView!

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift (hunk 3 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetCollectionCell/YeYa_PetCell.swift:12]: @IBOutlet private weak var breedLabel: UILabel!
    @IBOutlet private weak var ageLabel: UILabel!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        setupLayout()
    }

Corrected the order of methods

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift (hunk 4 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift:4]: final class YeYa_PetFilterCollectionViewCell: UICollectionViewCell {
    @IBOutlet private weak var petIconImage: UIImageView!

remove space

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift (hunk 5 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/PetFilterCollectionCell/YeYa_PetFilterCollectionViewCell.swift:18]: final class YeYa_PetFilterCollectionViewCell: UICollectionViewCell {
    @IBOutlet private weak var petIconImage: UIImageView!
    @IBOutlet private weak var kindOfPetLabel: UILabel!
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        setupLayout()
    }

Corrected the order of methods

File: UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/YeYa_PetsViewController.swift (hunk 6 above)
Comment (fimillion) [reply UIMentoringSession/MenteesChallenges/Pets Main Card UI Chellenge/YefimYakunin/YeYa_PetsViewController.swift:6]: @IBOutlet private weak var filtersCollectionView: UICollectionView!
    @IBOutlet private weak var petsCollectionView: UICollectionView!
    private var pets: [YeYa_PetModel] = []
    private var filters: [YeYa_PetFilterModel] = []
    private var filteredPets: [YeYa_PetModel] = []
    private var selectedFilterType: YeYa_PetType = .allPets

All variables are now private

**Comment by fimillion** [comment]
Thank you for the review )

**Pull request merged**